pyrobuf/parse_proto.py

```diff
@@ -209,17 +222,19 @@ def _parse_message(self, s, current, tokens, enums, imported_enums):
 
         for token in tokens:
             if token.token_type == 'MESSAGE':
-                current.messages[token.name] = self._parse_message(s, token, tokens, enums, imported_enums)
+                token.full_name = current.full_name + token.name
+                current.messages[token.name] = self._parse_message(s, token, tokens, messages, enums, imported_enums)
+                # updates the dictionary of known/parsed messages.
+                messages.update(current.messages)
```

Comment:
1. good point, I've implemented your suggestion.
2. that's right for scoping, we could perhaps introduce another ParserMessage member, like `ancestor_messages`. And when `token.token_type == 'MESSAGE'`, we could update it from `current.ancestor_messages`. That's just an idea, but I don't know if that would work. Also if we want to take care of scoping, we need to write a unit test, that shows we're not allowed to create outer scoped messages.